gnovm/pkg/gnolang/op_binary.go

```diff
@@ -475,6 +475,10 @@ func isEql(store Store, lv, rv *TypedValue) bool {
 				rfv.GetClosure(store)
 		}
 	case PointerKind:
+		if lv.T != rv.T {
```

Comment:
```suggestion
		if lv.T.TypeID() != rv.T.TypeID() {
```
What about using TypedID? Some of the current tests are failing.  Here is an example case of what will fail. It returns false instead of true.
```go
package bar

type Node struct{}

func CheckNil() bool {
	var n *Node
	return n.IsNil()
}

func (n *Node) IsNil() bool {
	return n == nil
}
```